gettsim/transfers/arbeitsl_geld.py

```diff
@@ -42,71 +45,147 @@ def arbeitsl_geld_m(
 
     Parameters
     ----------
-    arbeitsl_monate_lfdj
-        See basic input variable :ref:`arbeitsl_monate_lfdj <arbeitsl_monate_lfdj>`.
-    arbeitsl_monate_vorj
-        See basic input variable :ref:`arbeitsl_monate_vorj <arbeitsl_monate_vorj>`.
-    arbeitsl_monate_v2j
-        See basic input variable :ref:`arbeitsl_monate_v2j <arbeitsl_monate_v2j>`.
+    dauer_nach_versich_pfl
+        See :func:`dauer_nach_versich_pfl`.
+    dauer_nach_alter
+        See :func:`dauer_nach_alter`.
+    arbeitsl_geld_bezug_m
+        See basic input variable :ref:`arbeitsl_geld_bezug_m <arbeitsl_geld_bezug_m>`.
+
 
     Returns
     -------
 
     """
-    return arbeitsl_monate_lfdj + arbeitsl_monate_vorj + arbeitsl_monate_v2j
+    if anwartschaftszeit:
+        anspruchsdauer_gesamt = min(dauer_nach_alter, dauer_nach_versich_pfl)
+        out = max(anspruchsdauer_gesamt - arbeitsl_geld_bezug_m, 0)
+    else:
+        out = 0
+
+    return out
 
 
 def arbeitsl_geld_berechtigt(
-    arbeitsl_monate_gesamt: int,
     alter: int,
-    sum_ges_rente_priv_rente_m: float,
+    arbeitssuchend: bool,
+    restliche_anspruchsdauer: int,
     arbeitsstunden_w: float,
     arbeitsl_geld_params: dict,
+    geburtsjahr: int,
+    ges_rente_params: dict,
 ) -> bool:
     """Check eligibility for unemployment benefit.
 
-    Different rates for parent and non-parents. Take into account actual wages. There
-    are different replacement rates depending on presence of children
-
     Parameters
     ----------
-    arbeitsl_monate_gesamt
-        See :func:`arbeitsl_monate_gesamt`.
     alter
         See basic input variable :ref:`alter <alter>`.
-    sum_ges_rente_priv_rente_m
-        See basic input variable :ref:`sum_ges_rente_priv_rente_m
-        <sum_ges_rente_priv_rente_m>`.
+    arbeitssuchend
+        See basic input variable :ref:`arbeitssuchend <arbeitssuchend>`.
+    restliche_anspruchsdauer
+        See :func:`restliche_anspruchsdauer`.
     arbeitsstunden_w
         See basic input variable :ref:`arbeitsstunden_w <arbeitsstunden_w>`.
     arbeitsl_geld_params
         See params documentation :ref:`arbeitsl_geld_params <arbeitsl_geld_params>`.
+    geburtsjahr
+        See basic input variable :ref:`geburtsjahr <geburtsjahr>`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
 
     Returns
     -------
 
     """
+    regelaltersgrenze = ges_rente_regelaltersgrenze(geburtsjahr, ges_rente_params)
+
     out = (
-        (
-            arbeitsl_monate_gesamt
-            <= arbeitsl_geld_params["dauer_auszahlung"]["max_dauer"]
-        )
-        and (alter < arbeitsl_geld_params["altersgrenze"]["alter"])
-        and (sum_ges_rente_priv_rente_m == 0)
+        arbeitssuchend
+        and (restliche_anspruchsdauer > 0)
+        and (alter < regelaltersgrenze)
         and (arbeitsstunden_w < arbeitsl_geld_params["stundengrenze"])
     )
+
     return out
 
 
+def dauer_nach_alter(
```

Comment:
I think we do not need `dauer_nach_alter` and `dauer_nach_versich_pfl` as separate functions. Can put this code to the beginning of `restliche_anspruchsdauer`